demos/http/http_demo_s3_upload/http_demo_s3_upload.c

```diff
@@ -0,0 +1,626 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
```

Comment:
Maybe we can add `AWS IoT Device SDK for Embedded C V202009.00` to make is easier to change the version during release.